db/db.py

```diff
@@ -1,17 +1,17 @@
 import logging
 import sqlite3
-from typing import List, Literal
+from typing import List, Literal, Optional
 
 from .objects import Market, Order, Trade
 
 
 class Database:
-    def __init__(self, db: str):
-        self.db = db
-        self.conn, self.cursor = None, None
+    def __init__(self, conn: sqlite3.Connection):
+        self.conn = conn
+        self.conn.execute("PRAGMA foreign_keys = ON")
```

Comment:
also probably a good idea to turn on WAL mode too

Reply:
i think we can set this on database initialization though, no need for per-connection config